From: George McCollister <george.mccollister@gmail.com>
Date: Thu, 2 Dec 2010 09:31:38 -0600
Subject: [PATCH] check if splice() takes 6 arguments

checking for the presence of splice() isn't sufficient. Older
implimentations only accept 4 arguments while this version of glib
assumes that it takes 6.

Signed-off-by: George McCollister <george.mccollister@gmail.com>
[mol: configure.in -> configure.ac]
Signed-off-by: Michael Olbrich <m.olbrich@pengutronix.de>
---
 configure.ac |   16 +++++++++++++++-
 1 file changed, 15 insertions(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index 48868d2..f186882 100644
--- a/configure.ac
+++ b/configure.ac
@@ -970,7 +970,21 @@ AC_CHECK_FUNCS(lstat strerror strsignal memmove vsnprintf stpcpy strcasecmp strn
 AC_CHECK_FUNCS(chown lchmod lchown fchmod fchown link utimes getgrgid getpwuid)
 AC_CHECK_FUNCS(getmntent_r setmntent endmntent hasmntopt getfsstat getvfsstat)
 # Check for high-resolution sleep functions
-AC_CHECK_FUNCS(splice)
+AC_CACHE_CHECK([for splice()],
+  [ac_cv_func_splice],
+  [AC_LINK_IFELSE([AC_LANG_PROGRAM([[
+#define _GNU_SOURCE
+#include <stdlib.h>
+#include <fcntl.h>
+  ]],[[
+    splice(0, NULL, 0, NULL, (size_t)0, (unsigned int)0);
+  ]])],
+  [ac_cv_func_splice=yes],
+  [ac_cv_func_splice=no])])
+
+AS_IF([test "x$ac_cv_func_splice" = "xyes"],[
+  AC_DEFINE([HAVE_SPLICE],[1],[If we have splice])
+])
 AC_CHECK_FUNCS(prlimit)
 
 # To avoid finding a compatibility unusable statfs, which typically
